src/java/htsjdk/samtools/sra/SRAUnalignmentIterator.java

```diff
@@ -0,0 +1,168 @@
+*
+*                            PUBLIC DOMAIN NOTICE
+*               National Center for Biotechnology Information
+*
+*  This software/database is a "United States Government Work" under the
+*  terms of the United States Copyright Act.  It was written as part of
+*  the author's official duties as a United States Government employee and
+*  thus cannot be copyrighted.  This software/database is freely available
+*  to the public for use. The National Library of Medicine and the U.S.
+*  Government have not placed any restriction on its use or reproduction.
+*
+*  Although all reasonable efforts have been taken to ensure the accuracy
+*  and reliability of the software and data, the NLM and the U.S.
+*  Government do not and cannot warrant the performance or results that
+*  may be obtained by using this software or data. The NLM and the U.S.
+*  Government disclaim all warranties, express or implied, including
+*  warranties of performance, merchantability or fitness for any particular
+*  purpose.
+*
+*  Please cite the author in any work or product based on this material.
+*
+* ===========================================================================
+*
+*/
+
+package htsjdk.samtools.sra;
+
+import htsjdk.samtools.*;
+import ngs.*;
+
+import java.util.Iterator;
+
+/**
+ * Created by andrii.nikitiuk on 9/3/15.
+ */
+public class SRAUnalignmentIterator implements Iterator<SAMRecord> {
+    private ValidationStringency validationStringency;
+
+    private ReadCollection run;
+    private SAMFileHeader header;
+    private SRAIterator.RecordRangleInfo recordRangleInfo;
+
+    private ReadIterator unalignedIterator;
+    private boolean hasMoreUnalignedReads = true;
+    private Boolean hasMoreUnalignedFragments = false;
+    private int lastUnalignedFragmentIndex;
+
+    private SRALazyRecord lastRecord;
+
+    public static long getNumberOfReads(ReadCollection run) throws ErrorMsg {
+        return run.getReadCount(Read.all); // TODO: check if that is okay
+    }
+
+    public SRAUnalignmentIterator(final ReadCollection run, final SAMFileHeader header, SRAIterator.RecordRangleInfo recordRangleInfo, Chunk chunk) {
+        this.run = run;
+        this.header = header;
+        this.recordRangleInfo = recordRangleInfo;
+
+        long readStart = chunk.getChunkStart() - recordRangleInfo.getTotalReferencesLength();
+        if (readStart < 0) {
+            readStart = 0;
+        } else if (readStart >= recordRangleInfo.getNumberOfReads()) {
+            throw new RuntimeException("Invalid chunk provided: chunkStart position is after last read");
+        }
+
+        long readEnd = chunk.getChunkEnd() - recordRangleInfo.getTotalReferencesLength();
+        if (readEnd > recordRangleInfo.getNumberOfReads()) {
+            readEnd = recordRangleInfo.getNumberOfReads();
+        } else if (readEnd <= 0) {
+            throw new RuntimeException("Invalid chunk provided: chunkStart position is after last read");
```

Comment:
Is this error message correct? Should it be referring to the chunkEnd position instead?